pytorch_lightning/trainer/trainer.py

```diff
@@ -1382,8 +1381,3 @@ def _on_exception(self):
         # save a checkpoint for fault tolerant training. we don't use `log_dir` to minimize the chances of failure.
         file_path = os.path.join(self.default_root_dir, ".pl_auto_save.ckpt")
         self.save_checkpoint(file_path)
-
-    @contextmanager
-    def _evaluation_context(self) -> Generator:
```

Comment:
Can we keep the decorator without `torch.inference_mode()` ?

Reply:
Some users needed to override this to enable gradient computation on validation.

Reply:
hi @tchaton - for this PR i'd like to keep it as clean a revert as possible (undoing exactly #8813)

figuring out how computation can happen in validation could be a follow up. what do you think?